File: src/model_v4/entities/SBS_SW_BESTAND.ts
import { Column, Entity, Index, JoinColumn, ManyToOne, PrimaryColumn } from "typeorm";
import { SBS_SW } from "./SBS_SW";


@Entity("SBS_SW_BESTAND",{schema:"sbsdb"})
@Index("FKB99A81B33E4C6525",["sW_INDEX",])
export class SBS_SW_BESTAND {

    @PrimaryColumn("bigint",{ 
        nullable:false,
        primary:true,
        name:"SWBESTAND_INDEX"
        })
    SWBESTAND_INDEX:string;
        

    @Column("date",{ 
        nullable:true,
        default: () => "'NULL'",
        name:"ANSCH_DAT"
        })
    ANSCH_DAT:string | null;
        

    @Column("double",{ 
        nullable:true,
        default: () => "'NULL'",
        name:"ANSCH_WERT"
        })
    ANSCH_WERT:number | null;
        

    @Column("bigint",{ 
        nullable:false,
        name:"ANZAHL"
        })
    ANZAHL:string;
        

    @Column("varchar",{ 
        nullable:true,
        length:50,
        default: () => "'NULL'",
        name:"INV_NR"
        })
    INV_NR:string | null;
        

   
    @ManyToOne(type=>SBS_SW, SBS_SW=>SBS_SW.sBS_SW_BESTANDs,{ onDelete: 'RESTRICT',onUpdate: 'RESTRICT' })
    @JoinColumn({ name:'SW_INDEX'})
    sW_INDEX:SBS_SW | null;

}

File: SbsdbServer/sbsdbv4/model/SbsSwBestand.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace hb.SbsdbServer.sbsdbv4.model
{
    [Table("SBS_SW_BESTAND")]
    public partial class SbsSwBestand
    {
        [Column("SWBESTAND_INDEX", TypeName = "bigint(20)")]
        public long SwbestandIndex { get; set; }
        [Column("ANSCH_DAT", TypeName = "date")]
        public DateTime? AnschDat { get; set; }
        [Column("ANSCH_WERT")]
        public double? AnschWert { get; set; }
        [Column("ANZAHL", TypeName = "bigint(20)")]
        public long Anzahl { get; set; }
        [Column("INV_NR", TypeName = "varchar(50)")]
        public string InvNr { get; set; }
        [Column("SW_INDEX", TypeName = "bigint(20)")]
        public long? SwIndex { get; set; }

        [ForeignKey("SwIndex")]
        [InverseProperty("SbsSwBestand")]
        public virtual SbsSw SwIndexNavigation { get; set; }
    }
}
